stable/datadog/templates/container-agent.yaml

```diff
@@ -77,6 +77,8 @@
     {{- if .Values.datadog.systemProbe.enabled }}
     - name: SYSTEM_PROBE_CONFIG_SYSPROBE_SOCKET
       value: /sysprobe/var/run
+    - name: sysprobe-config
+      mountPath: {{ template "datadog.confPath" . }}/system-probe.yaml
```

Comment:
This wouldn’t work because `sysprobe-config` is a ConfigMap that could contain several files.
This will create a directory named `system-probe.yaml` and the ConfigMap will be mounted inside this directory.
We should use the `subPath` directive here to select which subpart of the ConfigMap needs to be mounted.

Another reason why this change isn’t enough is because the [unix socket path set in `system-probe.yaml`](https://github.com/helm/charts/blob/333eefddc9941b759bcc3ae58488a8a99413bf67/stable/datadog/templates/system-probe-configmap.yaml#L18) doesn’t match the path where the [unix socket is mounted inside this container](https://github.com/helm/charts/blob/333eefddc9941b759bcc3ae58488a8a99413bf67/stable/datadog/templates/container-agent.yaml#L109).